src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -91,11 +85,23 @@ struct OperationViewModel {
             }
         }
 
+        var signatureParams = filterParams(for: operation.signatureParameters ?? [], with: [.path, .uri])
+        var optionsParams = filterParams(for: operation.signatureParameters ?? [], with: [.header, .query, .body])
```

Comment:
You could also make `filterParams(for: [Parameter]?, ...)` then you can handle the nil case with a guard and not need the `?? []`.

Reply:
Good point.